e2e-tests/testdata/echo_client/ts/client.ts

```diff
@@ -780,7 +780,10 @@ type CallParameters = Omit<RequestInit, "method" | "body" | "headers"> & {
 }
 
 // AuthDataGenerator is a function that returns a new instance of the authentication data required by this API
-export type AuthDataGenerator = () => (echo.AuthParams | undefined)
+export type AuthDataGenerator = () =>
+  | echo.AuthParams
+  | undefined
+  | Promise<echo.AuthParams | undefined>;
```

Comment:
Can we move the `| undefined` case to the end? It's a bit unintuitive otherwise.